app/javascript/components/pages/SearchEdit.js

```diff
@@ -1,13 +1,371 @@
-import React from "react"
-import PropTypes from "prop-types"
+import React, { Component } from "react";
+import { Button, Col, Form, FormGroup, Input, Label, Row } from "reactstrap";
+import { Redirect } from "react-router-dom";
+
+import PropTypes from "prop-types";
+import cityData from "../TicityCities.js";
+import findCities from "../../helpers/find-cities";
+
 class SearchEdit extends React.Component {
-    render() {
-        return (
-            <React.Fragment>
-                <h3>Edit Search Page</h3>
-            </React.Fragment>
-        );
-    }
+  constructor(props) {
+    super(props);
+    searchKey: "";
```

Comment:
why is searchKey and searchValue class properties instead of being part of this this.state?

Reply:
because it worked, but based on your comment we have moved them into state and reverified the functionality is still working correctly